Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM --platform=$BUILDPLATFORM gradle:jdk17 as builder
+WORKDIR /build
+COPY build.gradle settings.gradle gradle.properties ./
+COPY .git .git
+COPY src src
+RUN gradle --no-daemon build
```

Comment:
Don't use the gradle base image. Use eclipse-temurin:21-alpine and call ./gradlew instead of gradle.

Reply:
The project doesn't build with java 21 though, and i can't use java 17 because eclipse-temurin doesn't have builds for other architectures for 17

Reply:
The github actions as well as myself are using Java 21 and it builds fine

Reply:
for me it ends in
`6.51 A problem occurred evaluating root project 'ViaProxy'.
56.51 > Could not create task ':java8Jar'.
56.51    > Failed to calculate the value of task ':compileJava' property 'javaCompiler'.
56.51       > Cannot find a Java installation on your machine matching this tasks requirements: {languageVersion=17, vendor=any, implementation=vendor-specific} for LINUX on x86_64.
56.51          > No locally installed toolchains match and toolchain download repositories have not been configured.`

Reply:
Adding gradle's toolchain resolver should fix that https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning